### PR TITLE
SCRD-5901 Use optional chaining when possible

### PR DESCRIPTION
**Pull request by eapodaca**
* Simplifies logic in many cases by replacing `&&` with `?.`

### CODE REVIEW

File: src/pages/topology/Networks.js [hunk 1]
```diff
@@ -68,10 +68,7 @@ class Networks extends Component {
 
         let net_list = [];
         for(const [net_name, net_data] of Object.entries(nw_groups[nw_group_name])) {
-          if (net_data['control_planes'][cp_name] &&
-              net_data['control_planes'][cp_name]['clusters'] &&
-              net_data['control_planes'][cp_name]['clusters'][cluster_name]) {
-
+          if (net_data['control_planes']?.[cp_name]?.['clusters']?.[cluster_name]) {
```
[the remark this answers is not in the document]
Comment (GarySmith) [reply src/pages/topology/Networks.js]: The latest (forced push) version, which is only visible via the 'force-pushed' link below, has no ability to make comments there. But in that version, it lost two of the `?.`, but really only the first was extra.

**Comment by GarySmith** [comment]
The latest (forced push) version, which is only visible via the 'force-pushed' link above, has no ability to 
make comments there. But in that version, it lost two of the ?. on line 71 of Networks.js, but really only the first was extra.

File: src/utils/RouteUtils.js [hunk 2]
```diff
@@ -46,5 +46,5 @@ export function redirectToLogin(forced = true) {
 
 export function wasRedirectedToLogin() {
   const location = createHistory().location;
-  return location.state && location.state['forcedRedirect'] === true;
+  return location?.state['forcedRedirect'] === true;
```
Comment (jeremy-moffitt) [inline src/utils/RouteUtils.js]: I think this should be `location.state?.['forcedRedirect']`

**Comment by jeremy-moffitt** [comment]
on the roles page (`http://localhost:2209/#/services/roles`)  got the following: 
```TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at ServicesPerRole.js:74```

**Comment by jeremy-moffitt** [comment]
whats odd is there is no change to that file, trying on master...

**Comment by jeremy-moffitt** [comment]
works fine on master...

**Comment by jeremy-moffitt** [comment]
now I'm really confused because the problem isnt reproducing

**Comment by jeremy-moffitt** [comment]
walked through the Day0 and Day2 UIs after a full clearout of old data, seemed fine throughout, looks good

**Pull request merged**